lib/matplotlib/artist.py

```diff
@@ -329,19 +329,29 @@ def get_window_extent(self, renderer=None):
         """
         Get the artist's bounding box in display space.
 
-        The bounding box' width and height are nonnegative.
+        The bounding box's width and height are non-negative.
 
         Subclasses should override for inclusion in the bounding box
         "tight" calculation. Default is to return an empty bounding
         box at 0, 0.
 
-        Be careful when using this function, the results will not update
-        if the artist window extent of the artist changes.  The extent
-        can change due to any changes in the transform stack, such as
-        changing the Axes limits, the figure size, or the canvas used
-        (as is done when saving a figure).  This can lead to unexpected
-        behavior where interactive figures will look fine on the screen,
-        but will save incorrectly.
+        .. warning ::
+
+          Be careful when using this function, the results will not update if
+          the artist window extent of the artist changes.
+
+          The extent can change due to any changes in the transform stack, such
+          as changing the Axes limits, the figure size, the canvas used (as is
+          done when saving a figure), or the DPI.
+
+          This can lead to unexpected behavior where interactive figures will
+          look fine on the screen, but will save incorrectly.
+
+          To get accurate results you may need to manually call
+          `matplotlib.figure.Figure.savefig` or
+          `matplotlib.figure.Figure.draw_without_rendering` to have Matplotlib
+          compute the rendered size.
```

Comment:
This is unclear to me. I feel, we are mixing two things here an it's too short to understand:

1. `draw_without_rendering` ensures the window_extent is up-to-date for the current transform stack. This resolves incorrect intermediate state due to lazy evalutation, e.g. of the layout.
2. How does `savefig` help here? Is this this the double save idea of https://github.com/matplotlib/matplotlib/issues/2831#issuecomment-2799571987. If so, that needs more explanation. Also I'm not sure on the generality of the solution https://github.com/matplotlib/matplotlib/issues/2831#issuecomment-2799852381.

Reply:
I left a comment over on #2831 for why in the case of vector backends we may need savefig (tl;dr: `print_pdf` changes the DPI).

Reply:
I still think this is too terse to be helpful and actionable for users. However, I won't block over this.

Decide yourself whether it's good enough for now or how much additional effort you want to put into this.